### PR TITLE
feat(Truncate): now supports an optional `description` property

### PR DESCRIPTION
**Pull request by lukelooker**
Adds new `description` option for `Truncate` and implements access to it via ListItem and descendants.

Screenshot:

image

## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main//CONTRIBUTING.md#developer-checklist)

- [x] ♿️ Accessibility addressed
- [x] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [ ] 👾 Browsers tested
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11

### CODE REVIEW

File: packages/components/src/List/ListItem.tsx [hunk 1]
```diff
@@ -68,9 +68,10 @@ const TruncateWrapper: FC<{
   color?: string
   fontSize?: FontSizes
   lineHeight?: FontSizes
-}> = ({ children, color, fontSize, lineHeight }) => (
+  truncateDetail?: string
+}> = ({ children, color, truncateDetail, fontSize, lineHeight }) => (
   <Text color={color} fontSize={fontSize} lineHeight={lineHeight}>
-    <Truncate>{children}</Truncate>
+    <Truncate detail={truncateDetail}>{children}</Truncate>
```
Comment (jhardy) [inline packages/components/src/List/ListItem.tsx:74]: Overall looks good to me, only one thing came to mind is that in other components like `ListItem` and `DataTableCell` text that appears below the content is called `description`, where detail in `ListItem` goes to the right.  Thoughts?

Comment (lukelooker) [reply packages/components/src/List/ListItem.tsx:74]: 🤦  Yes, that would make sense I suppose :) - I'll update

**Review by jhardy — APPROVED** [review]
👍

File: packages/components/src/List/ListItem.tsx [hunk 2]
```diff
@@ -115,8 +116,12 @@ export type ListItemProps = CompatibleHTMLProps<HTMLElement> &
     itemRole?: ListItemRole
     /**
      * If true, text children and description will be truncated if text overflows
+     * Specifying `detail` will cause truncation tooltip for label to _always_ be presented
+     *
+     * Text specified in `detail` property will be displayed below `label` in the tooltip
```
Comment (ElliotTheHuman) [inline packages/components/src/List/ListItem.tsx:121]:
```suggestion
     * Specifying `description` will cause the truncation tooltip for label to _always_ be presented
     *
     * Text specified in `description` property will be displayed below `label` in the tooltip
```

**Review by ElliotTheHuman — COMMENTED** [review]
Some minor name / JSDoc feedback. No blockers, but wanted to point them.

File: packages/components/src/Truncate/Truncate.tsx [hunk 3]
```diff
@@ -28,18 +28,27 @@ import styled from 'styled-components'
...
 export interface TruncateProps extends WidthProps {
   className?: string
+  /**
+   * Specifying `description` will cause truncation tooltip to _always_ be presented
+   * Text specified in `description` property will be displayed below `children` supplied
+   */
+  description?: string
 }
 
+export type TruncateProp = undefined | boolean | { description: string }
```
[the remark this answers is not in the document]
Comment (lukelooker) [reply packages/components/src/Truncate/Truncate.tsx]: That's fair - adjusting :)

**Review by ElliotTheHuman — APPROVED** [review]
LGTM!

**Pull request merged**